tests/testthat/test-daisyTimeline.R

```diff
@@ -0,0 +1,188 @@
+test_that("daisyTimeline works with data frame input", {
+  # Create test data frame
+  events_df <- data.frame(
+    date = c("2022", "2023", "2024"),
+    content = c("Event 1", "Event 2", "Event 3"),
+    stringsAsFactors = FALSE
+  )
+  
+  # Test function execution
+  result <- daisyTimeline(events_df)
+  
+  # Check that result is an htmlwidget
+  expect_s3_class(result, "htmlwidget")
+  expect_s3_class(result, "daisyTimeline")
+  
+  # Check that the data is properly formatted
+  expect_type(result$x$events, "list")
+  expect_length(result$x$events, 3)
+  
+  # Check first event structure
+  expect_equal(result$x$events[[1]]$date, "2022")
+  expect_equal(result$x$events[[1]]$content, "Event 1")
+  
+  # Check all events are properly converted
+  dates <- sapply(result$x$events, function(x) x$date)
+  contents <- sapply(result$x$events, function(x) x$content)
+  expect_equal(dates, c("2022", "2023", "2024"))
+  expect_equal(contents, c("Event 1", "Event 2", "Event 3"))
+})
```

Comment:
This can be simplified - and we shouldn't use `sapply()`.  Let's find a simpler way to do this.